python/paddle/fluid/tests/unittests/mkldnn/test_elementwise_add_mkldnn_op.py

```diff
@@ -15,121 +15,318 @@
         self.out = self.x + self.y
 
-    def init_kernel_type(self):
-        self.use_mkldnn = True
-
 
-class TestMKLDNNElementwiseAddOp_Vector(TestElementwiseAddOp_Vector):
-    def init_kernel_type(self):
-        self.use_mkldnn = True
+class TestMKLDNNElementwiseAddOp_Vector(TestMKLDNNElementwiseAddOp):
+    def init_input_output(self):
+        self.x = np.random.random((100, )).astype(self.dtype)
+        self.y = np.random.random((100, )).astype(self.dtype)
+        self.out = np.add(self.x, self.y)
 
 
-class TesMKLDNNtElementwiseAddOp_broadcast_0(TestElementwiseAddOp_broadcast_0):
+class TestMKLDNNElementwiseAddOp_broadcast_0(TestMKLDNNElementwiseAddOp):
     def init_input_output(self):
         self.x = np.random.rand(100, 2, 3, 4).astype(self.dtype)
         self.y = np.random.rand(100).astype(self.dtype)
         self.out = self.x + self.y.reshape(100, 1, 1, 1)
 
-    def init_kernel_type(self):
-        self.use_mkldnn = True
+    def init_axis(self):
+        self.axis = 0
 
 
-class TestMKLDNNElementwiseAddOp_broadcast_1(TestElementwiseAddOp_broadcast_1):
+class TestMKLDNNElementwiseAddOp_broadcast_1(TestMKLDNNElementwiseAddOp):
     def init_input_output(self):
         self.x = np.random.rand(2, 100, 3, 4).astype(self.dtype)
         self.y = np.random.rand(100).astype(self.dtype)
         self.out = self.x + self.y.reshape(1, 100, 1, 1)
 
-    def init_kernel_type(self):
-        self.use_mkldnn = True
+    def init_axis(self):
+        self.axis = 1
 
 
-class TestMKLDNNElementwiseAddOp_broadcast_2(TestElementwiseAddOp_broadcast_2):
+class TestMKLDNNElementwiseAddOp_broadcast_2(TestMKLDNNElementwiseAddOp):
+    def print_info(self):
+        print("### TestMKLDNNElementwiseAddOp_broadcast2 ###")
+
     def init_input_output(self):
-        self.x = np.random.rand(2, 2, 3, 100).astype(self.dtype)
+        self.x = np.random.rand(2, 3, 4, 100).astype(self.dtype)
         self.y = np.random.rand(100).astype(self.dtype)
         self.out = self.x + self.y.reshape(1, 1, 1, 100)
 
-    def init_kernel_type(self):
-        self.use_mkldnn = True
 
+class TestMKLDNNElementwiseAddOp_broadcast_3(TestMKLDNNElementwiseAddOp):
+    def print_info(self):
+        print("### TestMKLDNNElementwiseAddOp_bradcast_3 ###")
 
-class TestMKLDNNElementwiseAddOp_broadcast_3(TestElementwiseAddOp_broadcast_3):
-    def init_kernel_type(self):
-        self.use_mkldnn = True
+    def init_input_output(self):
+        self.x = np.random.rand(2, 30, 40, 5).astype(self.dtype)
+        self.y = np.random.rand(30, 40).astype(self.dtype)
+        self.out = self.x + self.y.reshape(1, 30, 40, 1)
 
+    def init_axis(self):
+        self.axis = 1
 
-class TestMKLDNNElementwiseAddOp_broadcast_4(TestElementwiseAddOp_broadcast_4):
-    def init_kernel_type(self):
-        self.use_mkldnn = True
 
+class TestMKLDNNElementwiseAddOp_broadcast_4(TestMKLDNNElementwiseAddOp):
+    def print_info(self):
+        print("### TestMKLDNNElementwiseAddOp_broadcast_4 ###")
```

Comment:
Where is print_info function used? Others look good to me.

Reply:
It is a leftover. I will remove it.

Reply:
Done.